docs/ADD_NEW_EXCHANGE.md

```diff
@@ -0,0 +1,1079 @@
+		[]byte(f.API.Credentials.Secret),
+	)
+	sign := crypto.HexEncodeToString(hmac)
+	req := Authenticate{Operation: "login",
+		Args: AuthenticationData{
+			Key:  f.API.Credentials.Key,
+			Sign: sign,
+			Time: intNonce,
+		},
+	}
+	return f.WebsocketConn.SendJSONMessage(req)
+}
+```
+
+- Create an unsubscribe function if the exchange has the functionality:
+
+```go
+// Unsubscribe sends a websocket message to stop receiving data from the channel
+func (f *FTX) Unsubscribe(channelToSubscribe wshandler.WebsocketChannelSubscription) error {
+	var unSub WsSub
+	a, err := f.GetPairAssetType(channelToSubscribe.Currency)
+	if err != nil {
+		return err
+	}
+	unSub.Operation = "unsubscribe"
+	unSub.Channel = channelToSubscribe.Channel
+	unSub.Market = f.FormatExchangeCurrency(channelToSubscribe.Currency, a).String()
+	return f.WebsocketConn.SendJSONMessage(unSub)
+}
+```
+
+- Complete websocket setup in wrapper:
+
+Add websocket functionality if supported to Setup:
+
+```go
+	err = f.Websocket.Setup(
+		&wshandler.WebsocketSetup{
+			Enabled:                          exch.Features.Enabled.Websocket,
+			Verbose:                          exch.Verbose,
+			AuthenticatedWebsocketAPISupport: exch.API.AuthenticatedWebsocketSupport,
+			WebsocketTimeout:                 exch.WebsocketTrafficTimeout,
+			DefaultURL:                       ftxWSURL,
+			ExchangeName:                     exch.Name,
+			RunningURL:                       exch.API.Endpoints.WebsocketURL,
+			Connector:                        f.WsConnect,
+			Subscriber:                       f.Subscribe,
+			UnSubscriber:                     f.Unsubscribe,
+			Features:                         &f.Features.Supports.WebsocketCapabilities,
+		})
+	if err != nil {
+		return err
+  }
+  ```
+
+Below are the features supported by FTX API protocol:
+
+  ```go
+  f.Features = exchange.Features{
+		Supports: exchange.FeaturesSupported{
+			REST:      true,
+			Websocket: true,
+			RESTCapabilities: protocol.Features{
+				TickerFetching:      true,
+				KlineFetching:       true,
+				TradeFetching:       true,
+				OrderbookFetching:   true,
+				AutoPairUpdates:     true,
+				AccountInfo:         true,
+				GetOrder:            true,
+				GetOrders:           true,
+				CancelOrders:        true,
+				CancelOrder:         true,
+				SubmitOrder:         true,
+				TradeFee:            true,
+				FiatDepositFee:      true,
+				FiatWithdrawalFee:   true,
+				CryptoWithdrawalFee: true,
+			},
+			WebsocketCapabilities: protocol.Features{
+				OrderbookFetching: true,
+				TradeFetching:     true,
+				Subscribe:         true,
+				Unsubscribe:       true,
+				GetOrders:         true,
+				GetOrder:          true,
+			},
+			WithdrawPermissions: exchange.NoAPIWithdrawalMethods,
+		},
+		Enabled: exchange.FeaturesEnabled{
+			AutoPairUpdates: true,
+		},
+	}
+```
+
+- Link websocket to wrapper functions:
+
+Initially the functions return nil or common.ErrNotYetImplemented
+
+```go
```

Comment:
This still requires the addition of 
```
// GetWebsocket returns a pointer to the exchange websocket
func (f *FTX) GetWebsocket() (*wshandler.Websocket, error) {
	return f.Websocket, nil
}
```
Otherwise things will error